lib/CSkillHandler.h

```diff
@@ -27,31 +28,42 @@ class DLL_LINKAGE CSkill // secondary skill
 		template <typename Handler> void serialize(Handler & h, const int version)
 		{
 			h & description;
+			if(version >= 785)
+			{
+				h & images;
+			}
 			h & effects;
 		}
 	};
 
 	std::vector<LevelInfo> levels; // bonuses provided by basic, advanced and expert level
 
 public:
-	CSkill(SecondarySkill id = SecondarySkill::DEFAULT);
+	CSkill(SecondarySkill id = SecondarySkill::DEFAULT, std::string identifier = "default");
 	~CSkill();
 
 	void addNewBonus(const std::shared_ptr<Bonus> & b, int level);
 	void setDescription(const std::string & desc, int level);
+	void setImages(const std::vector<std::string> & images, int level);
 	const std::vector<std::shared_ptr<Bonus>> & getBonus(int level) const;
 	const std::string & getDescription(int level) const;
+	const std::vector<std::string> & getImages(int level) const;
 	std::string toString() const;
 
 	SecondarySkill id;
 	std::string identifier;
 	std::string name; //as displayed in GUI
+	std::vector<int> gainChance; // gainChance[0/1] = default gain chance on level-up for might/magic heroes

```

Comment:
This should be two scalar fields, or std::array<int32_t, 2>.Also please note that fixed size type highly recommended for serializable fields.